tests/unit/errors.spec.ts

```diff
@@ -22,6 +22,23 @@ describe('Errors: ', () => {
         expect(custom.is).toHaveProperty('NoSuchEntity', true);
     });
 
+    it('should allow additional metadata fields', () => {
+        const error = errors.InvalidArgument;
+        const originDescription = error.description;
+        const origincode = error.code;
+        const origintype = error.type;
+        const custom = error.addMetadataEntry('fieldName', [{ key: 'value' }]);
+        expect(custom.metadata.get('fieldName')).toBeTruthy();
+        expect(custom.metadata.get('fieldName')).toEqual([{ key: 'value' }]);
+        // Other properties should not change
+        expect(custom.is).toHaveProperty('InvalidArgument', true);
```

Comment:
`expect(custom.is[origintype]).toBeTruthy()` a little more precise ? Not sure about that one, do as you feel it right.